projects/igniteui-angular/src/lib/grids/grid/grid-filtering-advanced.spec.ts

```diff
@@ -2659,18 +2684,26 @@ describe('IgxGrid - Advanced Filtering', () => {
     });
 });
 
-function selectColumnInEditModeExpression(fix, dropdownItemIndex: number) {
+function selectColumnInEditModeExpressionBase(fix, dropdownItemIndex: number) {
     GridFunctions.clickAdvancedFilteringColumnSelect(fix);
     fix.detectChanges();
     GridFunctions.clickAdvancedFilteringSelectDropdownItem(fix, dropdownItemIndex);
+}
+
+function selectColumnInEditModeExpression(fix, dropdownItemIndex: number) {
+    selectColumnInEditModeExpressionBase(fix, dropdownItemIndex);
     tick();
     fix.detectChanges();
 }
 
-function selectOperatorInEditModeExpression(fix, dropdownItemIndex: number) {
+function selectOperatorInEditModeExpressionBase(fix, dropdownItemIndex: number) {
```

Comment:
It seems the only difference between the **selectOperatorInEditModeExpression** and the **selectOperatorInEditModeExpressionBase** methods is the type of waiting (tick or await wait()).
I believe we can simply create a new additional method called **selectOperatorInEditModeExpressionAsync** (without changing the existing non-async one), which is absolutely analogical to the **selectOperatorInEditModeExpression**, but uses await wait(), rather than tick(). This way it would become obvious pretty fast the difference between the two and tests will not need to call await wait() on their own.